### PR TITLE
[GTW-6943] Add new pagination GET Payments endpoint

### PR DESCRIPTION
**Pull request by sam-bonheure-cko**
# Description of changes in PR

[//]: # 'Please add your description here'

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [ ] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs (will be handled by @jason-dantzer

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: abc_spec/paths/payments.yaml [hunk 1]
```diff
@@ -163,3 +163,53 @@ post:
             $ref: '#/components/schemas/ValidationError'
     '502':
       description: Bad gateway
+get:
+  security:
+    - ApiSecretKey: [ ]
+  tags:
+    - Payments
+  summary: Get payments
```
Comment (jason-dantzer-cko) [inline abc_spec/paths/payments.yaml:171]: 1. The endpoint should be named Get payment lists. 
2. Can we add a beta tag next to the endpoint title?

File: abc_spec/paths/payments.yaml [hunk 2]
```diff
@@ -163,3 +163,54 @@ post:
             $ref: '#/components/schemas/ValidationError'
     '502':
       description: Bad gateway
+get:
+  security:
+    - ApiSecretKey: [ ]
+  tags:
+    - Payments
+  summary: Get payment lists
+  operationId: getPayments
```
Comment (katy-hughes-cko) [inline abc_spec/paths/payments.yaml:172]: is this just an arbitrary id ? should this be called getPaymentLists too?

**Review by jason-dantzer-cko — APPROVED** [review]

File: abc_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    type: integer
+    description: The numbers of items to return
+    example: 10
+  skip:
+    type: integer
+    description: The number of results to skip
+    example: 10
+  total_count:
+    type: integer
+    description: The total number of payments retrieved (without taking into consideration skip and limit parameters)
```
Comment (david-chandler-cko) [inline abc_spec/components/schemas/Payments/PaymentPaged.yaml:13]: I think this is ambiguous... do we mean "The total number of **matching** payments _available_"?
Right now 'retrieved' looks like it means the number returned by the paged request.

File: abc_spec/paths/payments.yaml [hunk 4]
```diff
@@ -163,3 +163,54 @@ post:
...
+        minimum: 0
+        default: 0
+      required: false
+      description: The number of results to skip
+    - in: query
+      name: reference
+      schema:
+        type: string
+      required: true
+      description: A reference (such as an order ID) that you can use later to identify the payment.
```
Comment (david-chandler-cko) [inline abc_spec/paths/payments.yaml:201]:
```suggestion
      description: A reference (such as an order ID) that you can use later to identify the payment
```

File: nas_spec/paths/payments.yaml [hunk 5]
```diff
@@ -284,3 +284,56 @@ post:
...
+    - Payments
+  security:
+    - OAuth:
+        - gateway
+        - gateway:payment-details
+    - ApiSecretKey: [ ]
+  summary: Get payment lists
+  description: |
+    <span style="font-size: 14px; font-weight: 500; color: rgb(60, 4, 133); background-color: rgb(237, 222, 255); border-radius: 4px; padding: 2px 6px" >Beta</span><p></p>
+    Returns a list of all payments for your business. The results will be
```
Comment (david-chandler-cko) [inline nas_spec/paths/payments.yaml:298]:
```suggestion
    Returns a list of all matching payments for your business. The results will be
```

File: nas_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    type: integer
+    description: The numbers of items to return
+    example: 10
+  skip:
+    type: integer
+    description: The number of results to skip
+    example: 10
+  total_count:
+    type: integer
+    description: The total number of payments retrieved (without taking into consideration skip and limit parameters)
```
Comment (david-chandler-cko) [inline nas_spec/components/schemas/Payments/PaymentPaged.yaml:13]:
```suggestion
    description: The total number of matching payments available (without taking into consideration skip and limit parameters)
```

File: abc_spec/paths/payments.yaml [hunk 7]
```diff
@@ -163,3 +163,54 @@ post:
...
+get:
+  security:
+    - ApiSecretKey: [ ]
+  tags:
+    - Payments
+  summary: Get payment lists
+  operationId: getPayments
+  description: |
+    <span style="font-size: 14px; font-weight: 500; color: rgb(60, 4, 133); background-color: rgb(237, 222, 255); border-radius: 4px; padding: 2px 6px" >Beta</span><p></p>
+    Returns a list of all payments for your business. The results will be returned in reverse chronological order,
```
Comment (david-chandler-cko) [inline abc_spec/paths/payments.yaml:175]:
```suggestion
    Returns a list of all matching payments for your business. The results will be returned in reverse chronological order,
```

File: abc_spec/paths/payments.yaml (hunk 2 above)
Comment (katy-hughes-cko) [reply abc_spec/paths/payments.yaml:172]: @sam-bonheure-cko

Comment (jason-dantzer-cko) [reply abc_spec/paths/payments.yaml:172]: I don't know what that is, Sam might be better to answer this one

File: abc_spec/components/schemas/Payments/PaymentPaged.yaml (hunk 3 above)
Comment (jason-dantzer-cko) [reply abc_spec/components/schemas/Payments/PaymentPaged.yaml:13]: you are right, it is indeed "The total number of matching payments available" not the total ones returned

Comment (jason-dantzer-cko) [inline abc_spec/components/schemas/Payments/PaymentPaged.yaml:13]: Change this to match NAS "The total number of matching payments available (without taking into consideration skip and limit parameters)"

File: abc_spec/paths/payments.yaml (hunk 2 above)
Comment (sam-bonheure-cko) [reply abc_spec/paths/payments.yaml:172]: Not sure either, don't think this is visualized in the UI anywhere

**Review by jason-dantzer-cko — CHANGES_REQUESTED** [review]

**Review by jason-dantzer-cko — APPROVED** [review]

File: abc_spec/paths/payments.yaml [hunk 8]
```diff
@@ -163,3 +163,54 @@ post:
...
+get:
+  security:
+    - ApiSecretKey: [ ]
+  tags:
+    - Payments
+  summary: Get payment lists
+  operationId: getPayments
+  description: |
+    <span style="font-size: 14px; font-weight: 500; color: rgb(60, 4, 133); background-color: rgb(237, 222, 255); border-radius: 4px; padding: 2px 6px" >Beta</span><p></p>
+    Returns a list of all matching payments for your business. The results will be returned in reverse chronological order,
```
Comment (sebastian-garcia-cardona-cko) [inline abc_spec/paths/payments.yaml:175]:
```suggestion
    Returns a list of your business' payments that match the specified reference. Results are returned in reverse chronological order, with the most recent payments shown first.
```

File: abc_spec/paths/payments.yaml [hunk 9]
```diff
@@ -163,3 +163,54 @@ post:
...
+    showing the last made payment first.
```
Comment (sebastian-garcia-cardona-cko) [inline abc_spec/paths/payments.yaml:176]:
```suggestion
```

File: abc_spec/paths/payments.yaml [hunk 10]
```diff
@@ -163,3 +163,54 @@ post:
...
+  parameters:
+    - in: query
+      name: limit
+      schema:
+        type: integer
+        minimum: 1
+        maximum: 100
+        default: 10
+      required: false
+      description: The numbers of results to return
```
Comment (sebastian-garcia-cardona-cko) [inline abc_spec/paths/payments.yaml:187]:
```suggestion
      description: The numbers of results to retrieve
```

File: abc_spec/paths/payments.yaml [hunk 11]
```diff
@@ -163,3 +163,54 @@ post:
...
+        minimum: 0
+        default: 0
+      required: false
+      description: The number of results to skip
+    - in: query
+      name: reference
+      schema:
+        type: string
+      required: true
+      description: A reference (such as an order ID) that you can use later to identify the payment
```
Comment (sebastian-garcia-cardona-cko) [inline abc_spec/paths/payments.yaml:201]:
```suggestion
      description: A reference, such as an order ID, that can be used to identify the payment
```

File: nas_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 12]
```diff
@@ -0,0 +1,19 @@
+type: object
+properties:
+  limit:
+    type: integer
+    description: The numbers of items to return
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/PaymentPaged.yaml:5]:
```suggestion
    description: The numbers of results to retrieve
```

File: nas_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+    example: 10
+  skip:
+    type: integer
+    description: The number of results to skip
+    example: 10
+  total_count:
+    type: integer
+    description: The total number of matching payments available (without taking into consideration skip and limit parameters)
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/components/schemas/Payments/PaymentPaged.yaml:13]:
```suggestion
    description: The total number of results matching the payment reference. This number is unaffected by the values of the `skip` and `limit` parameters
```

File: nas_spec/paths/payments.yaml [hunk 14]
```diff
@@ -284,3 +284,56 @@ post:
...
+    - Payments
+  security:
+    - OAuth:
+        - gateway
+        - gateway:payment-details
+    - ApiSecretKey: [ ]
+  summary: Get payment lists
+  description: |
+    <span style="font-size: 14px; font-weight: 500; color: rgb(60, 4, 133); background-color: rgb(237, 222, 255); border-radius: 4px; padding: 2px 6px" >Beta</span><p></p>
+    Returns a list of all matching payments for your business. The results will be
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/paths/payments.yaml:298]:
```suggestion
    Returns a list of your business' payments that match the specified reference. Results are returned in reverse chronological order, with the most recent payments shown first.
```

File: nas_spec/paths/payments.yaml [hunk 15]
```diff
@@ -284,3 +284,56 @@ post:
...
+    returned in reverse chronological order, showing the last made payments first. You
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/paths/payments.yaml:299]:
```suggestion
```

File: nas_spec/paths/payments.yaml [hunk 16]
```diff
@@ -284,3 +284,56 @@ post:
...
+    can use the optional parameters below to skip or limit results.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/paths/payments.yaml:300]:
```suggestion
    You can use the optional parameters below to skip or limit results.
```

File: nas_spec/paths/payments.yaml [hunk 17]
```diff
@@ -284,3 +284,56 @@ post:
...
+  parameters:
+    - in: query
+      name: limit
+      schema:
+        type: integer
+        minimum: 1
+        maximum: 100
+        default: 10
+      required: false
+      description: The numbers of results to return
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/paths/payments.yaml:310]:
```suggestion
      description: The numbers of results to retrieve
```

File: nas_spec/paths/payments.yaml [hunk 18]
```diff
@@ -284,3 +284,56 @@ post:
...
+        minimum: 0
+        default: 0
+      required: false
+      description: The number of results to skip
+    - in: query
+      name: reference
+      schema:
+        type: string
+      required: true
+      description: A reference (such as an order ID) that you can use later to identify the payment.
```
Comment (sebastian-garcia-cardona-cko) [inline nas_spec/paths/payments.yaml:324]:
```suggestion
      description: A reference, such as an order ID, that can be used to identify the payment
```

Comment (sebastian-garcia-cardona-cko) [reply nas_spec/paths/payments.yaml:324]: I've modified the copy based on the previous wording, but is the reference here not the query parameter that's being used to retrieve the results?

If so, I can update the description.

File: abc_spec/paths/payments.yaml (hunk 11 above)
Comment (sebastian-garcia-cardona-cko) [reply abc_spec/paths/payments.yaml:201]: I've modified the copy based on the previous wording, but is the reference here not the query parameter that's being used to retrieve the results?

If so, I can update the description.

**Review by sebastian-garcia-cardona-cko — CHANGES_REQUESTED** [review]

File: abc_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 19]
```diff
@@ -0,0 +1,19 @@
+type: object
+properties:
+  limit:
+    type: integer
+    description: The numbers of items to return
```
[the remark this answers is not in the document]
Comment (sam-bonheure-cko) [reply abc_spec/components/schemas/Payments/PaymentPaged.yaml:5]: Good point this is limited to 100 for now in public beta. With a default of 10.

File: nas_spec/paths/payments.yaml (hunk 18 above)
Comment (jason-dantzer-cko) [reply nas_spec/paths/payments.yaml:323]: Currently only reference is used so we can have the description stating reference. If we add other query parameters later we would need to update this endpoints description

**Review by jason-dantzer-cko — APPROVED** [review]

File: abc_spec/components/schemas/Payments/PaymentPaged.yaml [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+    description: The number of results to retrieve
+    example: 10
+  skip:
+    type: integer
+    description: The number of results to skip
+    example: 10
+  total_count:
+    type: integer
+    description: The total number of results matching the payment reference. This number is unaffected by the values of the `skip` and `limit` parameters
+
```
Comment (sam-bonheure-cko) [inline abc_spec/components/schemas/Payments/PaymentPaged.yaml:14]: whitespace

**Review by sebastian-garcia-cardona-cko — APPROVED** [review]

**Pull request merged**